stan/math/prim/fun/hypergeometric_1F0.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef STAN_MATH_PRIM_FUN_HYPERGEOMETRIC_1F0_HPP
+#define STAN_MATH_PRIM_FUN_HYPERGEOMETRIC_1F0_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err/check_less.hpp>
+#include <stan/math/prim/fun/boost_policy.hpp>
+#include <boost/math/special_functions/hypergeometric_1F0.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns the Hypergeometric 1F0 function applied to the
+ * input arguments:
+ * \f$ _1F_0(a;;z) = \sum_{k=1}^{\infty}\frac{\left(a\right)_kz^k}{k!}\f$
+ *
+ * \f$ \frac{\partial _1F_0\left(a;;z\right)}{\partial a} =
+ *    -\left(1-z\right)^{-a}\log\left(1 - z\right) \f$
+ *
+ * \f$ \frac{\partial _1F_0\left(a;;z\right)}{\partial z} =
+ *    a\left(1-z\right)^{-a-1} \f$
+ *
+ * @tparam Ta Arithmetic type of 'a' argument
+ * @tparam Tz Arithmetic type of 'z' argument
+ * @param[in] a Scalar 'a' argument
+ * @param[in] z Scalar z argument
+ * @return Hypergeometric 1F0 function
+ */
+template <typename Ta, typename Tz, require_all_arithmetic_t<Ta, Tz>* = nullptr>
+double hypergeometric_1f0(const Ta& a, const Tz& z) {
```

Comment:
Is this the `(double, double) -> double` full template specification? (sorry... notation is tough: `Ta = double`, `Tz = double`)

If so, this implementation doesn't make sense as the function template for `hypergeomtric_1f0` and we should probably implement this as a normal function. What do you think?

Based on the other implementations, we're not actually taking the route of defining the function template in `prim` and writing template specialization in each of the folders. We're using SFINAE to knock out template matches instead. So I think that means we don't have a single function template any more.

Am I thinking about this correctly? Thoughts? 

(My concrete suggestion: change this function to not have any template parameters.)

Reply:
Since this is the base template, I'd suggest having the return type be `return_type<Ta, Tz>` instead of `double`. I'll add the code suggestion right now.

The reason for that is the base template should be a catch-all for everything.

Reply:
```suggestion
return_type<Ta, Tz> hypergeometric_1f0(const Ta& a, const Tz& z) {
```

Reply:
Something feels slightly off: the base template has 3 template parameters, but the specializations have 4? Was that intended? (Did I miss the actual base template declaration somewhere?)

Reply:
> Something feels slightly off: the base template has 3 template parameters, but the specializations have 4? Was that intended? (Did I miss the actual base template declaration somewhere?)

Sorry you've lost me, all of the implementations here only have two parameters?

Reply:
Are you counting the `require_*` elements here? The number of those don't have to be consistent between the implementations